src/openrct2/object/ObjectManager.cpp

```diff
@@ -352,42 +358,42 @@ class ObjectManager final : public IObjectManager
 
     Object* RepositoryItemToObject(const ObjectRepositoryItem* ori, std::optional<int32_t> slot = {})
     {
-        if (ori == nullptr)
-            return nullptr;
-
-        Object* loadedObject = ori->LoadedObject.get();
-        if (loadedObject != nullptr)
-            return loadedObject;
```

Comment:
Would be nice if you could revert it back to this, its a lot easier to understand imho.

Reply:
Could you comment on this, @IntelOrca ? I don’t have any strong feelings either way.

Reply:
You can revert it if you want. This is subjective though.

Reply:
The reason I asked is because you wrote it, and I’m quite sure you must have had a good reason to change it. I’m just porting your object changes from the NSF to develop.

Reply:
The code is still doing the same, it just bails early which is easier to understand if you read a function, they are basically the pre-conditions before even bothering with anything.

Reply:
Ah, I see. I’ll restore that bit.